src/test/java/qna/domain/UserRepositoryTest.java

```diff
@@ -0,0 +1,104 @@
+package qna.domain;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import qna.UnAuthorizedException;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+@DataJpaTest
+public class UserRepositoryTest {
+    @Autowired
+    UserRepository users;
+
+    @Test
+    @DisplayName("User 저장 테스트")
+    void save_user_test() {
+        User JAVAJIGI = UserTest.JAVAJIGI;
+        User savedUser = users.save(JAVAJIGI);
+        assertAll(
+                () -> assertThat(savedUser.getId()).isNotNull(),
+                () -> assertThat(savedUser.getUserId()).isEqualTo(JAVAJIGI.getUserId())
+        );
+    }
+
+    @Test
+    @DisplayName("UserId로 User 조회 테스트")
+    void find_by_user_id_test() {
+        User JAVAJIGI = UserTest.JAVAJIGI;
+        users.save(JAVAJIGI);
+        Optional<User> findUser = users.findByUserId(JAVAJIGI.getUserId());
+        assertThat(findUser.isPresent()).isTrue();
+    }
+
+    @Test
+    @DisplayName("ID가 같은 두 클래스의 일치 테스트")
+    void identity_test() {
+        final User user1 = users.save(new User("test", "test1", "테스트", "test@test.com"));
+        final User user2 = users.findById(user1.getId()).get();
+        assertThat(user1).isEqualTo(user2);
+    }
+
+    @Test
+    @DisplayName("이름 및 이메일 업데이트 테스트")
+    void name_and_email_update_test() {
+        final User JAVAJIGI = users.save(UserTest.JAVAJIGI);
+        User target = new User(JAVAJIGI.getUserId(), JAVAJIGI.getPassword(), "HAPPY", "HAPPY@test.com");
+        JAVAJIGI.update(JAVAJIGI, target);
+        assertThat(users.findByUserId(JAVAJIGI.getUserId()).get().getEmail()).isEqualTo("HAPPY@test.com");
+    }
```

Comment:
assertj 를 활용하시면 Optional 에 대해 메서드 체이닝 검증도 가능하니 참고만 해주세요 🙏
```suggestion
        JAVAJIGI.update(JAVAJIGI, target);
        assertThat(users.findByUserId(JAVAJIGI.getUserId()))
                .get()
                .extracting(User::getEmail)
                .isEqualTo("HAPPY@test.com");
    }
```